Algorithm.CSharp/DuplicateKeyExceptionAlgorithm.cs

```diff
@@ -0,0 +1,83 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Algorithm.Framework.Alphas;
+using QuantConnect.Algorithm.Framework.Portfolio;
+using System;
+using QuantConnect.Interfaces;
+using System.Collections.Generic;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Demonstrate old duplicate key exception when combining QC500 and MeanVarianceOptimizationPortfolioConstructionModel
+    /// </summary>
+    public class DuplicateKeyExceptionAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        public override void Initialize()
+        {
+            SetStartDate(2013, 10, 7);
+            SetEndDate(2013, 10, 11);
+            SetCash(25000);
+
+            UniverseSettings.Resolution = Resolution.Daily;
+
+            AddUniverse(Universe.Index.QC500);
+            SetAlpha(new ConstantAlphaModel(InsightType.Price, InsightDirection.Up, TimeSpan.FromDays(1), .1D, .1D));
+            SetPortfolioConstruction(new MeanVarianceOptimizationPortfolioConstructionModel());
+        }
+
+        public bool CanRunLocally => false;
```

Comment:
Thanks for the test!
Just a note, I'm much more in favor of `CanRunLocally => true;` tests since those can be executed by each dev locally while testing or a unit test just for `ReturnsSymbolData` in this case

Reply:
I replaced that regression test for a unit test (cc @mchandschuh).